couch2pg/src/setup.js

```diff
@@ -25,12 +26,37 @@ const createTimestampIndex = `
 CREATE INDEX IF NOT EXISTS saved_timestamp ON ${db.postgresTable}(saved_timestamp);
 `;
 
+
+const checkTableExists = async (client) => {
+  const tableExists = `SELECT 1 FROM ${db.postgresTable} LIMIT 1;`;
+
+  try {
+    await client.query(tableExists);
+    return true;
+  } catch (error) {
+    // "Undefined table" error code in PostgreSQL
+    if (error.code === '42P01') {
+      return false;
+    }
+    throw error;
+  }
+};
+
 export const createDatabase = async () => {
   const client = await db.getPgClient();
   await client.query(createSchema);
-  await client.query(createTable);
-  await client.query(createDeleteIndex);
-  await client.query(createTimestampIndex);
+
+  // create the table and indexes if they don't exist
+  // note that `CREATE INDEX IF NOT EXISTS` acquires a lock
+  // even if the index exists, so cannot rely on it here,
+  // have to actually check if the table exists in a separate query
+  const tableExists = await checkTableExists(client);
```

Comment:
Can return early here instead: 
```suggestion
if (await checkTableExists(client)) {
  return;
}
```